tee_supplicant/src/tee_supplicant.c

```diff
@@ -335,32 +343,54 @@ static int process_request(const struct device *dev)
 
 static void tee_supp_main(void *p1, void *p2, void *p3)
 {
-	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 	const struct device *dev = p1;
+	struct thread_arg *arg = (struct thread_arg *)p2;
 	int rc = 0;
 
+	if (!arg) {
+		LOG_ERR("Invalid argument");
+		return;
+	}
 	/* TODO: implement terminating supplicant's thread */
 	while (1) {
-		rc = process_request(dev);
+		rc = process_request(dev, arg);
 		if (rc) {
 			LOG_ERR("Failed to process request, rc = %d", rc);
 		}
 	}
+	if (arg->allocated) {
```

Comment:
Is there an use case when `arg->allocated` will be `false`?

Reply:
For now no such use case. I'll remove it